src/application-insights/azext_applicationinsights/custom.py

```diff
@@ -58,26 +57,45 @@ def create_or_update_component(cmd, client, application, resource_group_name, lo
                                              tags=tags, workspace_resource_id=workspace_resource_id,
                                              public_network_access_for_ingestion=public_network_access_for_ingestion,
                                              public_network_access_for_query=public_network_access_for_query)
-    return client.create_or_update(resource_group_name, application, component)
+    from ._client_factory import applicationinsights_mgmt_plane_client
+    client = applicationinsights_mgmt_plane_client(cmd.cli_ctx, api_version='2020-02-02-preview').components
+    try:
+        return client.create_or_update(resource_group_name, application, component)
+    except CloudError as ex:
+        message = " Please use `az feature register " \
+                  "--name AIWorkspacePreview --namespace microsoft.insights` to register the feature"
+        ex.error._message = ex.error._message + message
+        raise ex
 
 
 def update_component(cmd, client, application, resource_group_name, kind=None, workspace_resource_id=None,
                      public_network_access_for_ingestion=None, public_network_access_for_query=None):
-    existing_component = client.get(resource_group_name, application)
-    if kind:
-        existing_component.kind = kind
+    from ._client_factory import applicationinsights_mgmt_plane_client
+    existing_component = None
     if workspace_resource_id is not None:
+        latest_client = applicationinsights_mgmt_plane_client(cmd.cli_ctx, api_version='2020-02-02-preview').components
+        try:
+            existing_component = latest_client.get(resource_group_name, application)
+        except CloudError as ex:
+            message = " Please use `az feature register " \
+                      "--name AIWorkspacePreview --namespace microsoft.insights` to register the feature"
+            ex.error._message = ex.error._message + message
+            raise ex
         existing_component.workspace_resource_id = workspace_resource_id or None
```

Comment:
>workspace_resource_id [](start = 51, length = 21)

In which case, None will be assigned to existing_component.workspace_resource_id?